### PR TITLE
[refactor] remove types_test

### PR DESCRIPTION
**Pull request by yangzhg**
# Proposed changes

1. remove types_test, it will cause core dump in higher version gcc or
   clang, becuse of memory align, some code will be vectorize in higher
   gcc or clang
2. change string type length to 2GB instead of -1
3. modify unaccessable code

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java [hunk 1]
```diff
@@ -291,7 +291,8 @@ public static ScalarType createVarcharType(String lenStr) {
     public static ScalarType createStringType() {
         // length checked in analysis
         ScalarType type = new ScalarType(PrimitiveType.STRING);
-        type.len = -1;
+        // 2GB - 4
+        type.len = 2147483643;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java:295]: Define this magic number some where. Maybe in Type.java?
And explain why minus 4 in comment.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java [hunk 2]
```diff
@@ -69,6 +66,12 @@
     // Keep consistent with backend ColumnType::CHAR_INLINE_LENGTH
     public static final int CHAR_INLINE_LENGTH = 128;
 
+    // Max length of String types, in be storage layer store string length
+    // using int32, the max length is 2GB, the first 4 bytes store the length
+    // so the max available length is 2GB - 4
+
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java:72]: no need the space？

File: fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java [hunk 3]
```diff
@@ -69,6 +66,12 @@
...
+    public static final int MAX_STRING_LENGTH = 0x7fffffff - 4;
```
Comment (levy5307) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java:73]: I think `public static final int MAX_STRING_LENGTH = 2 << 30 - 1 - 4;` is more clear here. 
And you should explain why did you minus 1 here.

Comment (yangzhg) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java:73]: It is explained in comments that the value is 2GB - 4, why  use `public static final int MAX_STRING_LENGTH = 2 << 30 - 1 - 4;` ?  How to compute 2GB - 4 is not important and  2 << 30 does not have performance advantages

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**